common/generate_elastic_template.py

```diff
@@ -0,0 +1,96 @@
+
+from ._utils import _is_fraction_integer_or_half_integer
+
+
+def load_template(file_path: Path) -> str:
+    with open(file_path, "r") as file:
+        return file.read()
+
+
+template_file_path = Path(__file__).parent / "templates/elastic.template"
+with open(template_file_path, "r") as file:
+    elastic_input_template = file.read()
+
+
+def generate_elastic_template(
+    output_path: Path,
+    mass_t: float,
+    charge_t: float,
+    spin_t: Fraction,
+    mass_p: float,
+    charge_p: float,
+    spin_p: Fraction,
+    E_lab: float,
+    J_tot_min: Fraction,
+    J_tot_max: Fraction,
+    R_Coulomb: float,
+    reaction_name: str,
+    BE_t: float = 0.0,
+    R_match: float = 60.0,
+    step_size: float = 0.1,
+):
+    """
+    Generate an elastic scattering input template for Fresco.
+
+    Parameters:
+    output_path (Path): Path to save the generated template file.
+    mass_t (float): Mass of the target nucleus.
+    charge_t (float): Charge of the target nucleus.
+    spin_t (Fraction): Spin of the target nucleus (integer or half-integer).
+    mass_p (float): Mass of the projectile nucleus.
+    charge_p (float): Charge of the projectile nucleus.
+    spin_p (Fraction): Spin of the projectile nucleus (integer or half-integer).
+    E_lab (float): Laboratory energy of the projectile in MeV.
+    J_tot_min (Fraction): Minimum total angular momentum (integer or
+        half-integer).
+    J_tot_max (Fraction): Maximum total angular momentum (integer or
+        half-integer).
+    R_Coulomb (float): Coulomb radius in fm.
+    reaction_name (str): Name of the reaction for file naming.
+    BE_t (float): Binding energy of the target in MeV. Default is 0.0.
+    R_match (float): Matching radius in fm. Default is 60.0.
+    step_size (float): Step size for the radial mesh in fm. Default is 0
+    """
+
+    if J_tot_min > J_tot_max:
+        raise ValueError("J_tot_min cannot be greater than J_tot_max.")
+    if J_tot_min < 0 or J_tot_max < 0:
+        raise ValueError("J_tot_min and J_tot_max must be non-negative.")
+    if not _is_fraction_integer_or_half_integer(J_tot_min):
+        raise ValueError("J_tot_min must be an integer or half-integer.")
+    if not _is_fraction_integer_or_half_integer(J_tot_max):
+        raise ValueError("J_tot_max must be an integer or half-integer.")
+
+    # Define placeholder replacements
+    replacements = {
+        "HEADER": reaction_name,
+        "STEP_SIZE": str(step_size),
+        "RMATCH": str(R_match),
+        "J_TOT_MIN": str(float(J_tot_min)),
+        "J_TOT_MAX": str(float(J_tot_max)),
+        "E_LAB": str(E_lab),
+        "MASS_P": str(mass_p),
+        "CHARGE_P": str(charge_p),
+        "MASS_T": str(mass_t),
+        "CHARGE_T": str(charge_t),
+        "S_PROJECTILE": str(float(spin_p)),
+        "I_GROUND": str(float(spin_t)),
+        "E_GROUND": str(BE_t),
+        "COULOMB_R": str(R_Coulomb),
+    }
+
+    modified_template = elastic_input_template[:]
```

Comment:
Would it be better to just load the template file's contents anew here with each invocation?  That would preclude having to determine if any accidental changes to `elastic_input_template` in one invocation could lead to wrong output in a subsequent invocation (i.e., we don't have to think about persistence issues for that "constant" variable within the scope of this file).

Would any extra performance hit prohibit some of your use cases?